### PR TITLE
Update firebase_database_instance docs and samples for region

### PR DESCRIPTION
**Pull request by fredzqm**
Design Doc: [go/rtdb-terraform-modules](https://goto.google.com/rtdb-terraform-modules)
Tracking Bug: [b/244215914](http://b/244215914) RTDB Terraform modules

RTDB is only available in 3 regions today: `us-central1`, `europe-west1`, `asia-southeast1`.
https://firebase.google.com/docs/projects/locations#rtdb-locations

The default region (first region launched) is `us-central1`. The docs page shows `us-west1`, which RTDB has no support in.

This PR updates docs and samples only. No code changes.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
Updates description for firebase_database_instance.region
```

### CODE REVIEW

File: mmv1/products/firebasedatabase/api.yaml
Comment (rainshen49) [inline mmv1/products/firebasedatabase/api.yaml:45]: nit: maybe make the link like markdown? `Check all [available regions](https://firebase.google.com/docs/projects/locations#rtdb-locations)`

**Review by rainshen49 — APPROVED** [review]

**Comment by fredzqm** [comment]
@rainshen49 ready to be submitted. Thanks!

**Review by rainshen49 — APPROVED** [review]
LGTM. I'm not actually a maintainer, so we'll wait for @ScottSuarez to merge

**Comment by fredzqm** [comment]
@ScottSuarez Finally got the tests to pass. They are a bit flaky now.

Please take a look. This PR changes no impl, mainly fixing the doc page.

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**
